webrender/src/display_list_flattener.rs

```diff
@@ -1029,6 +1029,21 @@ impl<'a> DisplayListFlattener<'a> {
             None
         };
 
+        // preserve-3d's semantics are to hoist all your children to be your siblings
+        // when doing backface-visibility checking, so we need to grab the backface-visibility
+        // of the lowest ancestor which *doesn't* preserve-3d, and AND it in with ours.
+        //
+        // No this isn't obvious or clear, it's just what we worked out over a day of testing.
```

Comment:
I'm sorry, I don't understand why we are looking for the nearest Flat ancestor. Is there an actual reason backing this logic, or just empirical evidence?